subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/DependencyInsightReportTask.java

```diff
@@ -173,11 +189,32 @@ public boolean isShowSinglePathToDependency() {
      *
      * @since 4.9
      */
-    @Option(option = "singlepath", description = "Show at most one path to each dependency")
+    @Option(option = "single-path", description = "Show at most one path to each dependency")
     public void setShowSinglePathToDependency(boolean showSinglePathToDependency) {
         this.showSinglePathToDependency = showSinglePathToDependency;
     }
 
+    @Internal
+    public boolean isShowingAllVariants() {
+        return showingAllVariants;
+    }
+
+    @Option(option = "all-variants", description = "Show all variants of each dependency")
+    public void setShowingAllVariants(boolean showingAllVariants) {
+        this.showingAllVariants = showingAllVariants;
+    }
```

Comment:
All new properties need to use the Provider API, so this should be...

1. a `Property<Boolean>` and be exposed with a single getter. The `@Option` can be on that getter.
2. the convention for this property should be set in `HelpTasksPlugin`
3. It could be marked as `@Console` (I think it doesn't quite matter if it's `@Console`, `@Internal` or `@Input` since this has no outputs)

We also need to have Javadoc on this property and a new entry in `src/docs/dsl/org.gradle.api.tasks.diagnostics.DependencyInsightReportTask.xml`